verilog/formatting/formatter.cc

```diff
@@ -266,12 +267,65 @@ verible::util::Status Formatter::Format(const ExecutionControl& control) {
   const auto unwrapped_lines =
       MakeUnwrappedLinesWorklist(*format_tokens_partitions, style_);
 
+  if (control.show_token_partition_policy) {
+    control.Stream() << "Token partition policy:\n";
+    for (const auto& uwline : unwrapped_lines) {
+      control.Stream() << " :: " << uwline << " :: " << uwline.PartitionPolicy() << "\n";
+    }
+    return verible::util::OkStatus();
+  }
+
+  std::vector<UnwrappedLine> fitted_lines;
+  fitted_lines.reserve(unwrapped_lines.size());
+  int fitted_lines_indent = -1;
+  for (const auto& uwline : unwrapped_lines) {
+    if ((uwline.PartitionPolicy() == PartitionPolicyEnum::kAppendFittingSubpartitions) && (fitted_lines.size() > 0)) {
+      const auto& p1 = *(std::prev(fitted_lines.end()));
+      const auto& p2 = uwline;
+
+      if (p1.TokensRange().back().TokenEnum() == '(') {
```

Comment:
I would avoid having to perform checks on specific token enums in this phase.  It should ideally remain unaware of contents apart from what their minimum occupied space would be.  Better isolation of roles/knowledge.

Reply:
Fixed that.